### ISSUE TITLE
member-ordering fails for bound methods

### ISSUE THREAD

**Issue by j-f1** [nzakas/eslint-plugin-typescript#45]
With the default configuration, this fails:

```ts
class Foo {
  constructor () {}
  method () {}
  boundMethod = () => {}
}
```

It claims that the property `boundMethod` should be placed before the constructor. To fix this, the code in `getNodeType` should check if the value of the property is a `FunctionExpression` or `ArrowFunctionExpression`, then assume that it is a method.

**Comment by weirdpattern** [comment]
I assume I should also check type literals and interfaces, in which case is not a `FunctionExpression` or `ArrowFunctionExpression`, but a `TSFunctionType`

```ts
type Foo = {
  // interpreted as a property (TSPropertySignature), but processed as a method
  boundMethod = () => {}
}
```

Working on this right now...

**Comment by weirdpattern** [comment]
We can probably even include new options to sort these type of properties... for now I will concentrate in fixing this, then we can discuss expanding the rule even further...
